cmd/new_test.go

```diff
@@ -0,0 +1,263 @@
+	uniqueProjectName := filepath.Base(tempdir)
+
+	var args = newArgs{
+		interactive:       false,
+		name:              uniqueProjectName,
+		prompt:            promptForValue,
+		secretsProvider:   "default",
+		templateNameOrURL: "typescript",
+	}
+
+	err := runNew(args)
+	assert.NoError(t, err)
+
+	removeStack(t)
+
+	proj := loadProject(t, tempdir)
+	assert.Equal(t, uniqueProjectName, proj.Name.String())
+}
+
+func TestCreatingProjectWithEnteredName(t *testing.T) {
+	tempdir, _ := ioutil.TempDir("", "test-env")
+	defer os.RemoveAll(tempdir)
+	assert.NoError(t, os.Chdir(tempdir))
+	uniqueProjectName := filepath.Base(tempdir)
+
+	promptMock := func(
+		yes bool, valueType string, defaultValue string, secret bool,
+		isValidFn func(value string) error, opts display.Options) (string, error) {
+		return uniqueProjectName, nil
+	}
+
+	var args = newArgs{
+		interactive:       false,
+		name:              uniqueProjectName,
+		templateNameOrURL: "typescript",
+		prompt:            promptMock,
+		secretsProvider:   "default",
+	}
+
+	err := runNew(args)
+	assert.NoError(t, err)
+
+	removeStack(t)
+
+	proj := loadProject(t, tempdir)
+	assert.Equal(t, uniqueProjectName, proj.Name.String())
+}
+
+const projectName = "test_project"
+
+func promptMock(name string) promptForValueFunc {
+	return func(yes bool, valueType string, defaultValue string, secret bool,
+		isValidFn func(value string) error, opts display.Options) (string, error) {
+		if valueType == "project name" {
+			err := isValidFn(projectName)
+			return name, err
+		}
+		return "", nil
+	}
+}
+
+func TestCreatingProjectWithExistingSpecifiedNameFails(t *testing.T) {
+	tempdir, _ := ioutil.TempDir("", "test-env")
+	defer os.RemoveAll(tempdir)
+	assert.NoError(t, os.Chdir(tempdir))
+
+	backendInstance = &backend.MockBackend{
+		DoesProjectExistF: func(ctx context.Context, name string) (bool, error) {
+			return name == projectName, nil
+		},
+	}
+
+	var args = newArgs{
+		interactive:       false,
+		name:              projectName,
+		prompt:            promptForValue,
+		secretsProvider:   "default",
+		templateNameOrURL: "typescript",
+	}
+
+	err := runNew(args)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "project with this name already exists")
+}
+
+func TestCreatingProjectWithExistingEnteredNameFails(t *testing.T) {
+	tempdir, _ := ioutil.TempDir("", "test-env")
+	defer os.RemoveAll(tempdir)
+	assert.NoError(t, os.Chdir(tempdir))
+
+	backendInstance = &backend.MockBackend{
+		DoesProjectExistF: func(ctx context.Context, name string) (bool, error) {
+			return name == projectName, nil
+		},
+	}
+
+	var args = newArgs{
+		interactive:       true,
+		prompt:            promptMock(projectName),
+		secretsProvider:   "default",
+		templateNameOrURL: "typescript",
+	}
+
+	err := runNew(args)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "project with this name already exists")
+}
+
+func TestGeneratingProjectWithExistingSpecifiedNameSucceeds(t *testing.T) {
```

Comment:
The distinction between "Creating" vs. "Generating" is easy to miss. Wondering if this should be named something like `TestCreatingGenerateOnlyProject...` or something like that.